main.go

```diff
@@ -95,16 +105,30 @@ func realMain(logger hclog.Logger) error {
 	}
 	defer listener.Close()
 
-	config, err := rest.InClusterConfig()
+	cfg, err := rest.InClusterConfig()
 	if err != nil {
 		return err
 	}
-	clientset, err := kubernetes.NewForConfig(config)
+	clientset, err := kubernetes.NewForConfig(cfg)
 	if err != nil {
 		return err
 	}
 
-	srv := providerserver.NewServer(serverLogger, flags, clientset)
+	namespace, err := os.ReadFile(namespaceFile)
+	if err != nil {
+		return fmt.Errorf("failed to read namespace from file: %w", err)
+	}
+	hmacSecretSpec := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      *&flags.HMACSecretName,
```

Comment:
Does this need to be dereferenced and then referenced for some reason?

```suggestion
			Name:      flags.HMACSecretName,
```

Reply:
Oops, thanks for spotting that.